src/Command/CreateEntity/User/BulkUserCreationCommand.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+/*
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2021 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Command\CreateEntity\User;
+
+use InvalidArgumentException;
+use OAT\SimpleRoster\Service\Bulk\BulkCreateUsersService;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Throwable;
+
+class BulkUserCreationCommand extends Command
+{
+    public const NAME = 'roster:create:user';
+
+    private BulkCreateUsersService $bulkCreateUsersService;
+
+    private const DEFAULT_BATCH_SIZE = '100';
```

Comment:
shouldn't it be an int?

Reply:
Changed DEFAULT_BATCH_SIZE type to int but the batch size received from CLI is string . To make batch size type as unique the DEFAULT_BATCH_SIZE is converted as string in configure() -> $this->addOption(
            self::OPTION_BATCH_SIZE, and  pass it to  initializeBatchOption()